src/Workspaces/Core/Portable/Rename/IRemoteRenamer.cs

```diff
@@ -0,0 +1,208 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.FindSymbols;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Remote;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Rename
+{
+    internal interface IRemoteRenamer
+    {
+        Task<SerializableRenameLocations> FindRenameLocationsAsync(
+            PinnedSolutionInfo solutionInfo, SerializableSymbolAndProjectId symbol, SerializableRenameOptionSet options, CancellationToken cancellationToken);
+    }
+
+    internal struct SerializableRenameOptionSet
+    {
+        public bool RenameOverloads;
```

Comment:
for serializable objects do we have to leave the data fields mutable? This is supposed to be immutable data right?

Reply:
yes, it's a limitation of the serialization framework.

Reply:
boo :(